github/dependency_graph.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+type DependencyGraphService service
+
+// Sbom represents software bill of materials, which descibes the
+// packages/libraries that a repository depends on.
+type Sbom struct {
+	Sbom *SbomInfo `json:"sbom,omitempty"`
+}
+
+// When was the SBOM created and who created it
+type CreationInfo struct {
+	Created  *time.Time `json:"created,omitempty"`
+	Creators []*string  `json:"creators,omitempty"`
```

Comment:
This should be `[]string`, not `[]*string`.

Reply:
I'm new to Go. Could you please explain why?

Reply:
Might as well club all my questions, I saw usage of %v when formatting the API URI instead of %s like this:

`fmt.Sprintf("repos/%v/%v/dependency-graph/sbom", owner, repo)`

Is there a reason for this?

Reply:
https://github.com/google/go-github/issues/2563#issuecomment-1313753582

Pointers are used for structs and are not needed for strings.

`%v` is typically used in idiomatic Go unless there is a good reason not to, such as when passing a `[]byte`.

Reply:
Understood. Thank you!